src/NHibernate/Linq/Functions/StringGenerator.cs

```diff
@@ -95,15 +98,18 @@ public override HqlTreeNode BuildHql(MethodInfo method, Expression targetObject,
 
 	public class EndsWithGenerator : BaseHqlGeneratorForMethod
 	{
+		private static readonly MethodInfo MethodWithComparer = ReflectHelper.GetMethodDefinition<string>(x => x.EndsWith(null, default(StringComparison)));
+
 		public EndsWithGenerator()
 		{
-			SupportedMethods = new[] { ReflectHelper.GetMethodDefinition<string>(x => x.EndsWith(null)) };
+			SupportedMethods = new[] {ReflectHelper.GetMethodDefinition<string>(x => x.EndsWith(null)), MethodWithComparer,};
 		}
 
 		public override bool AllowsNullableReturnType(MethodInfo method) => false;
 
 		public override HqlTreeNode BuildHql(MethodInfo method, Expression targetObject, ReadOnlyCollection<Expression> arguments, HqlTreeBuilder treeBuilder, IHqlExpressionVisitor visitor)
 		{
+			LogIgnoredStringComparisonParameter(method, MethodWithComparer);
```

Comment:
Easier to check that function has 2 arguments

Reply:
>Easier to check that function has 2 arguments

But less "future-proof", we cannot assume another overload with 2 arguments may never be added into .Net.